Code/Tools/AssetProcessor/native/resourcecompiler/rcjob.cpp

```diff
@@ -542,6 +546,13 @@ namespace AssetProcessor
                                         builderParams.m_rcJob->GetJobEntry().m_pathRelativeToWatchFolder.toUtf8().data(), builderParams.m_rcJob->GetJobKey().toUtf8().data(),
                                         builderParams.m_rcJob->GetPlatformInfo().m_identifier.c_str(), builderParams.m_rcJob->GetOriginalFingerprint());
                                 }
+                                else
+                                {
+                                    AZStd::for_each(result.m_outputProducts.begin(), result.m_outputProducts.end(), [](auto& product)
+                                    {
+                                        product.m_outputFlags |= AssetBuilderSDK::ProductOutputFlags::CachedAsset;
+                                    });
```

Comment:
Is there a benefit to using for_each instead of a for range loop?

Reply:
not really, I can switch it to a for loop